### PR TITLE
Simplify high-level linting logic

### PR DESCRIPTION
**Pull request by gylaz**
The collaboration between `StyleChecker` and `Linter::Collection`
introduced additional complexity.

The new implementation feels easier to understand and is less code.

### CODE REVIEW

**Comment by teoljungberg** [comment]
Some background - the idea of `Linter::Collection` was to act as a facade for all of the linters we were going to use. But I think it's fair `StyleChecker` has that responsibility.

File: app/models/hound_config.rb [hunk 1]
```diff
@@ -9,6 +21,10 @@ class HoundConfig
 
   attr_reader_initialize :commit
 
+  def self.linter_names
```
Comment (teoljungberg) [inline app/models/hound_config.rb]: Could this be a constant instead?

File: app/models/hound_config.rb [hunk 2]
```diff
@@ -27,7 +43,7 @@ def fail_on_violations?
   private
 
   def default_config
-    Linter::Collection.linter_names.each.with_object({}) do |name, config|
+    self.class.linter_names.each.with_object({}) do |name, config|
```
Comment (teoljungberg) [inline app/models/hound_config.rb]: I prefer writing the class name instead of `self.class`. Your preference.

**Comment by teoljungberg** [comment]
I really like this refactoring!

File: app/models/hound_config.rb (hunk 1 above)
Comment (gylaz) [inline app/models/hound_config.rb]: Good idea. Were you thinking:
```rb
LINTER_NAMES = LINTERS.map { |klass| klass.name.demodulize.underscore }
```
or listing them out like `BETA_LINTERS`?

Comment (teoljungberg) [inline app/models/hound_config.rb]: I thought of the former, which gives us a nice bit of flexiblity.

**Comment by teoljungberg** [comment]
This LGTM, but I'd be happy to hear what @thorncp and @jcmorrow thinks

**Pull request merged**